modules/skybrowser/skybrowsermodule.cpp

```diff
@@ -0,0 +1,692 @@
+    }
+}
+
+void SkyBrowserModule::loadImages(const std::string& root, 
+                                  const std::filesystem::path& directory) {
+    _dataHandler->loadImages(root, directory);
+}
+
+int SkyBrowserModule::nLoadedImages() {
+    return _dataHandler->nLoadedImages();
+}
+
+void SkyBrowserModule::handleMouseClick(const MouseButton& button) {
+    setSelectedBrowser(_mouseOnPair->browserId());
+
+    if (button == MouseButton::Left) {
+        _startMousePosition = _mousePosition;
+        _startDragPosition = _mouseOnPair->selectedScreenSpacePosition();
+
+        // If it's not resize mode, it's drag mode
+        _interactionMode = MouseInteraction::Drag;
+        
+        // If target is clicked, it should unlock
+        if (_mouseOnPair->isTargetSelected()) {
+            _mouseOnPair->unlock();
+        }
+    }
+    // Fine tuning mode of target
+    else if (button == MouseButton::Right && _mouseOnPair->isBrowserSelected()) {
+        // If you start dragging around on the browser, the target unlocks
+        _mouseOnPair->unlock();
+        // Change view (by moving target) within browser if right mouse 
+        // click on browser
+        _startMousePosition = _mousePosition;
+        _startDragPosition = _mouseOnPair->getTarget()->screenSpacePosition();
+        _interactionMode = MouseInteraction::FineTune;
+    }
+}
+
+const std::unique_ptr<WwtDataHandler>& SkyBrowserModule::getWwtDataHandler() const {
+    return _dataHandler;
+}
+
+std::vector<std::unique_ptr<TargetBrowserPair>>& SkyBrowserModule::getPairs() {
+    return _targetsBrowsers;
+}
+
+int SkyBrowserModule::nPairs() {
+    return static_cast<int>(_targetsBrowsers.size());
+}
+
+TargetBrowserPair* SkyBrowserModule::getPair(const std::string& id) {
+    auto it = std::find_if(
+        _targetsBrowsers.begin(), 
+        _targetsBrowsers.end(),
+        [&](const std::unique_ptr<TargetBrowserPair>& pair) {
+            bool foundBrowser = pair->browserId() == id;
+            bool foundTarget = pair->targetId() == id;
+            return foundBrowser || foundTarget;
+        }
+    );
+    if (it == std::end(_targetsBrowsers)) {
+        return nullptr;
+    }
+    else {
+        return it->get();
+    }
+}
+
+void SkyBrowserModule::startRotatingCamera(glm::dvec3 endAnimation) {
+    // Save coordinates to rotate to in galactic world coordinates
+    _endAnimation = endAnimation;
+    _startAnimation = skybrowser::cameraDirectionGalactic();
+    _isCameraRotating = true;
+}
+
+void SkyBrowserModule::incrementallyRotateCamera(double deltaTime) {
+
+    // Find smallest angle between the two vectors
+    double angle = skybrowser::angleBetweenVectors(_startAnimation, _endAnimation);
+
+    if(angle > StopAnimationThreshold) {
+        
+        glm::dmat4 rotMat = skybrowser::incrementalAnimationMatrix(
+            _startAnimation, 
+            _endAnimation, 
+            deltaTime, 
+            AnimationSpeed
+        );
+
+        // Rotate
```

Comment:
Think about what comments are actually useful and what are redundant. This one seems redundant to me

remember that comments are another part of the code that has to be kept up to date if the implementation changes in the future, and they are often forgotten